components/twist/twist.app.js

```diff
@@ -0,0 +1,128 @@
+const events = require("./event-types.js");
+const axios = require("axios");
+
+module.exports = {
+  type: "app",
+  app: "twist",
+  propDefinitions: {
+    workspace: {
+      type: "string",
+      label: "Workspace",
+      description: "The workspace to watch for new events.",
+      optional: true,
+      async options() {
+        const workspaces = await this.getWorkspaces();
+        return workspaces.map((workspace) => {
+          return {
+            label: workspace.name,
+            value: workspace.id,
+          };
+        });
+      },
+    },
+    channel: {
+      type: "string",
+      label: "Channel",
+      description: "The channel to watch for new events.",
+      optional: true,
+      async options(opts) {
+        if (!opts.workspace)
+          return [{ label: "No Channels Found", value: "none" }];
```

Comment:
This is an interesting edge case, where a prop that depends on another one and the prop it depends on is missing or empty.

@dylburger food for thought